src/controller/CommissioningDelegate.h

```diff
@@ -447,38 +449,53 @@ struct ReadCommissioningInfo
     GeneralCommissioningInfo general;
 };
 
-struct AdditionalErrorInfo
+struct AttestationErrorInfo
 {
-    AdditionalErrorInfo(Credentials::AttestationVerificationResult result) : attestationResult(result) {}
+    AttestationErrorInfo(Credentials::AttestationVerificationResult result) : attestationResult(result) {}
     Credentials::AttestationVerificationResult attestationResult;
 };
 
+struct CommissioningErrorInfo
+{
+    CommissioningErrorInfo(app::Clusters::GeneralCommissioning::CommissioningError result) : commissioningError(result) {}
+    app::Clusters::GeneralCommissioning::CommissioningError commissioningError;
+};
+
+struct NetworkCommissioningStatusInfo
+{
+    NetworkCommissioningStatusInfo(app::Clusters::NetworkCommissioning::NetworkCommissioningStatus result) :
+        networkCommissioningStatus(result)
+    {}
+    app::Clusters::NetworkCommissioning::NetworkCommissioningStatus networkCommissioningStatus;
+};
+
 class CommissioningDelegate
 {
 public:
     virtual ~CommissioningDelegate(){};
     /* CommissioningReport is returned after each commissioning step is completed. The reports for each step are:
      * kReadCommissioningInfo - ReadCommissioningInfo
-     * kArmFailsafe: none
-     * kConfigRegulatory: none
+     * kArmFailsafe: CommissioningErrorInfo if there is an error
+     * kConfigRegulatory: CommissioningErrorInfo if there is an error
      * kSendPAICertificateRequest: RequestedCertificate
      * kSendDACCertificateRequest: RequestedCertificate
      * kSendAttestationRequest: AttestationResponse
-     * kAttestationVerification: AdditionalErrorInfo if there is an error
+     * kAttestationVerification: AttestationErrorInfo if there is an error
      * kSendOpCertSigningRequest: CSRResponse
      * kGenerateNOCChain: NocChain
      * kSendTrustedRootCert: None
      * kSendNOC: none
-     * kWiFiNetworkSetup: none
+     * kWiFiNetworkSetup: NetworkCommissioningStatusInfo if there is an error
      * kThreadNetworkSetup: none
-     * kWiFiNetworkEnable: none
+     * kWiFiNetworkEnable: NetworkCommissioningStatusInfo if there is an error
      * kThreadNetworkEnable: none
```

Comment:
```suggestion
     * kThreadNetworkEnable: NetworkCommissioningStatusInfo if there is an error
```
Again, pretty sure the callback that was modified above is used by both wifi and thread.